packages/nestjs-query/src/dataProvider/index.ts

```diff
@@ -419,30 +419,38 @@ const dataProvider = (client: GraphQLClient): Required<DataProvider> => {
       return (client as any).url; // url field in GraphQLClient is private
     },
     custom: async ({ url, method, headers, meta }) => {
-      if (url) {
-        client.setEndpoint(url);
-      }
+      const SUPPORTED_METHODS = ["get", "post"];
+      const requestUrl = url || (client as any).url;
 
-      if (headers) {
-        client.setHeaders(headers);
+      if (!SUPPORTED_METHODS.some((it) => it === method)) {
+        throw Error(`GraphQL does not support ${method} method.`);
       }
 
+      const validMethod = method as "get" | "post";
+
+      const _client = new GraphQLClient(requestUrl, {
+        method: validMethod,
+        headers,
+      });
+
       const gqlOperation = meta?.gqlMutation ?? meta?.gqlQuery;
 
       if (gqlOperation) {
-        const response: any = await client.request(
-          gqlOperation,
-          meta?.variables ?? {},
-        );
+        const response: any = await _client.request<BaseRecord>({
+          document: gqlOperation,
+          variables: meta?.variables,
+          requestHeaders: headers,
+        });
 
         return { data: response };
       }
 
       if (meta?.rawQuery) {
-        const response = await client.request<BaseRecord>(
-          meta.rawQuery,
-          meta.variables,
-        );
+        const response = await _client.request<BaseRecord>({
+          document: meta.rawQuery,
+          variables: meta.variables,
+          requestHeaders: headers,
```

Comment:
Oops, those aren't needed
```suggestion
          requestHeaders: headers,
```